docs/providers/aws/guide/serverless.yml.md

```diff
@@ -149,7 +149,8 @@ provider:
       executionLogging: true # Optional configuration which enables or disables execution logging. Defaults to true.
       level: INFO # Optional configuration which specifies the log level to use for execution logging. May be set to either INFO or ERROR.
       fullExecutionData: true # Optional configuration which specifies whether or not to log full requests/responses for execution logging. Defaults to true.
-      role: arn:aws:iam::123456:role # Optional IAM role for ApiGateway to use when managing CloudWatch Logs
+      role: arn:aws:iam::123456:role # Optional IAM role for ApiGateway to use when managing CloudWatch Logs. If 'role' is not configured and ensureRoleAssigned is true, a new role is created automatically.
+      ensureRoleAssigned: true # Specifies whether to ensure that the IAM role indicated in the 'role' property is configured in API Gateway CloudWatch log role ARN.
```

Comment:
As currently creation of role is a default mechanism (and to not introduce a breaking change, we should not change it) I think we should rather offer an opt-out mechanism.

Maybe let's add support for `accessManagedExternally` option, which if set to true will turn off any logic that tries to setup a role and assign it.

Reply:
Not sure if you've read the code before this comment, but that ensureRoleAssigned option *is* and opt-out mechanism (when true, does what it did until now, and this is the default value, and when false, it skips creating and/or checking the role).

Reply:
Maybe the comment should be more general? Something like:
```
# Specifies whether to ensure that ApiGateway has a CloudWatch Logs role assigned. Defaults to true
```

Reply:
I tweaked the comment of both role and ensureRoleAssigned to be slightly less confusing (even if they're less complete), see updated commit.

Reply:
> but that `ensureRoleAssigned option is an opt-out mechanism (when true, does what it did until now, and this is the default value, and when false, it skips creating and/or checking the role).

Yes, I know that. Still good convention is to have `false` as default in case of such option (so we turn on alternative handling by existence of a _truthy_ option and not by existence of _falsy_ option), it's less confusing that way.

Reply:
Ok, guess I was looking at all the other options in restApi with defaults to true and went along with it.

`roleExternallyManaged` sounds good?

Reply:
Changed it to roleManagedExternally defaulting to false.